### PR TITLE
Navigation API extracted to the extension

### PR DESCRIPTION
**Pull request by dryganets**
Replace Navigator API with extension.

We moved to react-native-deprecated-custom-components for StandardNavigationDelegate and to reactxp-experimental-navigation for ExperimentalNavigationDelegate.

There is a breaking change in the standard navigator. Styles should be passed as objects to the renderScene method.

https://github.com/facebook/react-native/commit/febf3d00ed228391ee4dd6541b0d2746dc56d21f

### CODE REVIEW

**Comment by dryganets** [comment]
Please don't merge it.
I still need to backport examples we have to react-navigation.

File: extensions/navigation/package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    "react": "^0.14.8",
+    "react-dom": "^15.4.1",
+    "react-native": "^0.42.0"
+  },
+  "devDependencies": {
+    "reactxp": "^0.42.0-rc.25",
+    "typescript": "2.4.1",
+    "tslint": "^5.0.0"
+  },
+  "author": "Sergei Dryganets",
```
Comment (berickson1) [inline extensions/navigation/package.json]: "ReactXP Team <reactxp@microsoft.com>"

File: extensions/navigation/src/web/Navigator.tsx [hunk 2]
```diff
@@ -10,17 +10,24 @@
 * pop, which update the state and cause transitions.
 */
 
-import _ = require('./utils/lodashMini');
+import _ = require('lodash');
```
Comment (berickson1) [inline extensions/navigation/src/web/Navigator.tsx]: I still think we want lodashmini for best bundled size

File: extensions/navigation/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    "compilerOptions": {
+        "declaration": true,
+        "noResolve": false,
+        "jsx": "react",
+        "module": "commonjs",
+        "target": "es5",
+        "experimentalDecorators": true,
+        "noImplicitAny": true,
+        "outDir": "dist/",
+        "lib": [
```
Comment (berickson1) [inline extensions/navigation/tsconfig.json]: if it's easy, please try enabling strictNullChecks here

File: extensions/navigation/package.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    "url": "https://github.com/Microsoft/reactxp/tree/master/extensions/navigation"
+  },
+  "keywords": [
+    "react-native",
+    "reactxp",
+    "react"
+  ],
+  "dependencies": {
+    "@types/lodash": "4.14.62",
+    "@types/node": "6.0.65",
```
Comment (berickson1) [inline extensions/navigation/package.json]: use semver '^' throughout so consumers don't get stuck on a particular version

File: extensions/navigation/tsconfig.json (hunk 3 above)
Comment (dryganets) [inline extensions/navigation/tsconfig.json]: I fixed it for react-native. For the web, it is way more tricky, and I don't have enough context for it now.

**Comment by dryganets** [comment]
@vinils, thank you fixed hello-world and removed the last reference from todo list.

File: extensions/navigation/src/common/Types.ts [hunk 5]
```diff
@@ -0,0 +1,143 @@
+// Use only for type data
```
Comment (berickson1) [inline extensions/navigation/src/common/Types.ts]: Missing copyright header

File: extensions/navigation/src/native-common/NavigatorExperimentalDelegate.tsx [hunk 6]
```diff
@@ -12,59 +12,67 @@
...
+    NavigatorRoute,
+    NavigatorState 
+} from '../common/Types';
 
 type NavigationSceneRendererProps = Navigation.NavigationSceneRendererProps;
 type NavigationState = Navigation.NavigationState;
 type NavigationRoute = Navigation.NavigationRoute;
 type NavigationTransitionProps = Navigation.NavigationTransitionProps;
-
+
```
Comment (berickson1) [inline extensions/navigation/src/native-common/NavigatorExperimentalDelegate.tsx]: nit: extra whitespace

File: extensions/navigation/src/web/Navigator.tsx [hunk 7]
```diff
@@ -10,17 +10,24 @@
...
+import _ = require('../common/lodashMini');
 import React = require('react');
 import ReactDOM = require('react-dom');
 import rebound = require('rebound');
 
 import { NavigatorSceneConfigFactory } from './NavigatorSceneConfigFactory';
 import { NavigatorSceneConfig }  from './NavigatorSceneConfigFactory';
-import RX = require('../common/Interfaces');
-import Styles from './Styles';
+import RX = require('reactxp');
```
Comment (berickson1) [inline extensions/navigation/src/web/Navigator.tsx]: nit: these should go up with other ambient imports

**Pull request merged**